source/ShopPanel.cpp

```diff
@@ -514,7 +514,7 @@ void ShopPanel::DrawShip(const Ship &ship, const Point &center, bool isSelected)
 
 	// Draw the ship name.
 	const Font &font = FontSet::Get(14);
-	const string &name = ship.Name().empty() ? ship.ModelName() : ship.Name();
+	const string &name = ship.Name().empty() ? (ship.VariantName().empty() ? ship.ModelName() : ship.VariantName()) : ship.Name();
```

Comment:
```suggestion
	const string &name = ship.Name().empty() ? (ship.VariantName()) : ship.Name();
```
Ship.VariantName() already does this: `return variantName.empty() ? modelName : variantName;` so thats not necessary

Reply:
Oh, that's great! I don't think we need the extra brackets in this case, though.

Reply:
Actually shouldn't we do one better, and do this inside of Name()?
Like, return the VariantName if the shipname is empty, the same way we do it for VariantName and the modelname?

Reply:
No, cause it should be possible to have a ship be without a name.